internal/controller/etcd/reconcile_status.go

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcd
+
+import (
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	ctrlutils "github.com/gardener/etcd-druid/internal/controller/utils"
+	"github.com/gardener/etcd-druid/internal/health/status"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/go-logr/logr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// mutateEtcdFn is a function which mutates the status of the passed etcd object
+type mutateEtcdFn func(ctx component.OperatorContext, etcd *druidv1alpha1.Etcd, logger logr.Logger) ctrlutils.ReconcileStepResult
+
+func (r *Reconciler) reconcileStatus(ctx component.OperatorContext, etcdObjectKey client.ObjectKey) ctrlutils.ReconcileStepResult {
+	etcd := &druidv1alpha1.Etcd{}
+	if result := r.getLatestEtcd(ctx, etcdObjectKey, etcd); ctrlutils.ShortCircuitReconcileFlow(result) {
+		return result
+	}
+	sLog := r.logger.WithValues("etcd", etcdObjectKey, "operation", "reconcileStatus").WithValues("runID", ctx.RunID)
+	originalEtcd := etcd.DeepCopy()
+	mutateETCDStepFns := []mutateEtcdFn{
+		r.mutateETCDStatusWithMemberStatusAndConditions,
+		r.inspectStatefulSetAndMutateETCDStatus,
+	}
+	for _, fn := range mutateETCDStepFns {
+		if stepResult := fn(ctx, etcd, sLog); ctrlutils.ShortCircuitReconcileFlow(stepResult) {
+			return stepResult
+		}
+	}
+	if err := r.client.Status().Patch(ctx, etcd, client.MergeFrom(originalEtcd)); err != nil {
+		sLog.Error(err, "failed to update etcd status")
+		return ctrlutils.ReconcileWithError(err)
+	}
+	return ctrlutils.ContinueReconcile()
+}
```

Comment:
It was moved to a different file because there is no predicate check for status updates. For deletion and reconcile spec handling only the predicate checks are done in the `reconciler.go`. From that perspective this does not break consistency.

Reply:
Makes sense, thanks.